### DISCUSSION TITLE
fheroes2 keeps muting in Win10 volume mixer

### DISCUSSION THREAD

**Discussion by Mickula** [ihhub/fheroes2#4655] [Q&A] [answered]
fheroes2 keeps muting in Win10 volume mixer. I try restart PC, reinstall of fheroes, reinstall of audio drivers and Win10 troubleshooter. Any ideas?

**Comment by oleg-derevenetz** [comment, marked as answer]
Hi @Mickula,

1. What type of music do you have - MIDI or OGG?
2. What volume for music and sounds have you selected?
3. When exactly volume mixer is muted - right after fheroes2 startup (so you cannot hear, say, music in a main menu) or after some specific actions in the game (e.g. battle)?

**Comment by Mickula** [comment]
1. No difference whatever i choose.
2. Same as 1.
3. Right after startup.

p.s.: i have same issue on two PCs

**Reply by oleg-derevenetz** [reply]
@Mickula So you have a non-zero volume in the game settings both for music and sound, you start the game and the Windows mixer is immediately muted for this game (and there is no sound in the whole game, including the main menu)?

**Reply by Mickula** [reply]
Yes, just like you wrote.

**Reply by Mickula** [reply]
p.s.: When i ALT TAB from game and move bar in mixer up, sound starts, but when i go back to the game window, it immediately stops.

**Reply by oleg-derevenetz** [reply]
@Mickula All the symptoms looks like you have music turned off in the game, like this:

![Music_off](https://user-images.githubusercontent.com/32623900/142444957-89e8d6f7-f0d3-4f4e-a458-20016aa1ca37.jpg)

In this case sound volume will be set to 0 in the Windows sound mixer each time the game window is activated. When the game is in background, the sound volume in the Windows sound mixer is always zero, because all game sounds are muted in the background.

**Comment by Mickula** [comment]
After update of Windows i have no more problems with mixer, but i really appreciate your help.

p.s.: I know this is a different topic, but is in current version of fheroes2 (0.9.9) chain lightning?

**Comment by oleg-derevenetz** [comment]
@Mickula

> p.s.: I know this is a different topic, but is in current version of fheroes2 (0.9.9) chain lightning?

Yes, and not only in 0.9.9, it was here a long ago before it.

![Chain-Lightning](https://user-images.githubusercontent.com/32623900/142701216-81e4cb95-85fe-499a-acbb-df47aa482801.jpg)

**Reply by Mickula** [reply]
Is in game some cheats or debug menu for test all spells?

**Reply by oleg-derevenetz** [reply]
@Mickula There is the "BATTLE ONLY" mode in the "NEW GAME" menu, in this mode you can choose the hero, edit hero's army, parameters and spell book contents and then perform a battle against another hero (either AI-controlled or human-controlled).

**Comment by a1exsh** [comment]
> Is in game some cheats or debug menu for test all spells?

One of the debug flags in the config file gives you a debug hero, which, I believe, has all spells in his book.
